### ISSUE TITLE
Gauntlet Dark Legacy: scaled dithering broken when cropping

### ISSUE THREAD

**Issue by BParks21** [PCSX2/pcsx2#11447]
### Describe the Bug

Starting with build v1.7.5911 scaled dithering in fullscreen when cropping is broken. Causes vertical bands. Every version from v1.7.5910 and back the scaled dithering is perfectly fine under these conditions. Most games that use dithering work fine and do not exhibit this issue. The banding seems to occur at 8x internal resolution but scaled dithering looks broken at other resolutions just without banding.  v1.7.5910 and back the scaled dithering looks good at all internal resolutions with cropping. 

### Reproduction Steps

Set the game to scaled dithering, run at a non native res (specifically 8x), crop all sides of the screen, go fullscreen then run the character back and forth.

### Expected Behavior

Scaled dithering should look fine with cropping like previous versions. 

### PCSX2 Revision

v1.7.5911 and up

### Operating System

Windows 11

### CPU

i7 8086k

### GPU

Nvidia GTX 1080ti

### GS Settings

Crop all sides, scaled dithering, 8x internal resolution. 

### Emulation Settings

default

### Logs & Dumps

Video: 
https://drive.google.com/file/d/15bYUah-kzhjHIH_JR0Ln08k9BygO5vF5/view?usp=drive_link

Muti Frame GS Dump:
https://drive.google.com/file/d/1zA5fXQrkqvHPIblnnuDR8aj0yLaGioC0/view?usp=drive_link

Savestate:
[SLUS-20047 (15030637).01.p2s.zip](https://github.com/user-attachments/files/15943904/SLUS-20047.15030637.01.p2s.zip)

**Comment by BParks21** [comment]
I'm using a 1440p display when seeing the binding, on a 4k screen it happens at 4x resolution in fullscreen. Afain with previous builds all internal resolutions the scaled branding looks fine even on the 4k display.

**Comment by TheTechnician27** [comment]
[Scooby-Doo! Night of 100 Frights_SLUS-20349_20240623110121.zip](https://github.com/user-attachments/files/15945243/Scooby-Doo.Night.of.100.Frights_SLUS-20349_20240623110121.zip) Night of 100 Frights has a seemingly small SW mips issue here.

**Comment by lightningterror** [comment]
@BParks21 Tested v2.1.68 and I can't reproduce the issue.

**Comment by Dreadmoth** [comment]
After 1.7.5910, for some reason scaled dithering looks identical to unscaled dithering in this game:

[1.7.5910 scaled dithering](https://github.com/user-attachments/assets/51f6d892-8b1a-44cb-b98f-bbc9453ab3ae)
[1.7.5910 unscaled dithering](https://github.com/user-attachments/assets/ba3c144a-0176-4023-bf57-570cc2651e9b)
[2.1.69 scaled dithering](https://github.com/user-attachments/assets/3274cd2e-fd47-44fa-ab46-21e6c56bd04b)

In other games I've tried that use dithering (e.g. Destroy All Humans! 2), the options seem to behave as intended.

---

I think the banding behaviour is a result of the bilinear scaling of internal resolution → window / display resolution disagreeing with the unscaled dithering pattern. Changing cropping and IR just creates different moiré patterns.

The banding can be mitigated by enabling 4xRGSS downsampling (Settings → Graphics → Post-Processing → TV Shader):
[2.1.69 dithering + 4xRGSS](https://github.com/user-attachments/assets/543c86b7-01be-4345-bb94-810dcf355145)

...or eliminated by using the "Force 32bit" dithering mode (albeit with brighter / less accurate-to-PS2 colours):
[2.1.69 Force 32-bit](https://github.com/user-attachments/assets/f8f9688e-2426-4f36-ab1a-d757c61cd6e3)

**Comment by refractionpcsx2** [comment]
if it's using native scaling, that could cause the dithering to act like native.

**Comment by BParks21** [comment]
Yeah I can still easily produce it. Like @Dreadmoth said scaled dithering also looks basically identical to unscaled with this game which isn't the case from 1.7.5910 and back. Scaled dithering used to work correctly with this game.

**Comment by BParks21** [comment]
Seems other games are affected the same way as Gauntlet. Also some are affected like Gauntlet but only when using unscaled dithering. Not sure if this is helps much.

**Comment by lightningterror** [comment]
You can also post gs dumps of them as well.

**Comment by Mrlinkwii** [comment]
whats the status of this ?

**Comment by BParks21** [comment]
The same. Driver Parllel Lines also suffers from this issue. Strange it's not all dithered games.

**Comment by Mrlinkwii** [comment]
is this still an issue ?

**Comment by BParks21** [comment]
I can still produce it the same way.

**Comment by Dreadmoth** [comment]
> is this still an issue ?

Scaled and unscaled dithering still appear to be identical in the provided dump:

[2.9.23 scaled](https://github.com/user-attachments/assets/bb4ae39e-1a18-42c8-962c-a72e353fa2c5)

[2.9.23 unscaled](https://github.com/user-attachments/assets/77a40925-4b65-4438-8ddb-447e6fb34c4c)
